### ISSUE TITLE
AniDB Posters Sometimes Fail to Download on Daily

### ISSUE THREAD

**Issue by natyusha** [ShokoAnime/ShokoServer#1045]
## VERSION INFORMATION

Latest (https://github.com/ShokoAnime/ShokoServer/commit/253918fa65702019677d4f4494567f37122e22bd)

## LOG FILE
An excerpt from the log is included as the rest isn't relevant.
```
23:05:03| CommandRequest_DownloadAniDBImages --- Saving new CommandRequest: DownloadAniDBImages CommandRequest_DownloadImage_15683_False
Images Queue state change: Downloading Image  - AniDB Posters: 15683
23:05:03| CommandRequest_GetReleaseGroupStatus --- Saving new CommandRequest: AniDB_GetReleaseGroupStatus CommandRequest_GetReleaseGroupStatus_15683
23:05:03| CommandRequest_DownloadAniDBImages --- Processing CommandRequest_DownloadAniDBImages: 15683
23:05:03| CommandRequest_DownloadAniDBImages --- Images failed to download: Can't find characters for anime with ID: 15683
General Queue state change: Getting group status info from UDP API for Anime: 15683
.
23:07:05| ShokoServiceImplementationImage --- Could not find AniDB_Cover image: /home/shoko/.shoko/Shoko.CLI/images/AniDB/15/251092.jpg
23:07:05| ShokoServiceImplementationImage --- Could not find AniDB_Cover image: /home/shoko/.shoko/Shoko.CLI/images/AniDB/15/251092.jpg
23:07:05| ShokoServiceImplementationImage --- Could not find AniDB_Cover image: /home/shoko/.shoko/Shoko.CLI/images/AniDB/15/251092.jpg
```

## DESCRIPTION

On the latest version of daily server AniDB series posters will fail to download under certain circumstances.

## STEPS TO REPRODUCE

In order for this to occur two things need to happen:

1. Character and/or Creator images must be enabled in Shoko Server's Settings -> AniDB -> Download Options.
2. The AniDB series in question must be missing Cast and/or Staff images **OR** the Cast and/or Staff images must be blank.

A few examples of series that meet the above criteria are linked below:

https://anidb.net/anime/1177
https://anidb.net/anime/1326
https://anidb.net/anime/15683

The commit which is causing this issue is likely located [here](https://github.com/ShokoAnime/ShokoServer/commit/c47431f276f5996c290690c90beb1c7e333c9807#diff-dc11547b7f20be267f26eb7282dff0d41c06cb6ed1807f6361700259db4079e7). From a cursory look at the code it seems that when the `FailedToDownloadNoID` `LogWarning` triggers for the creators or staff the poster will not download.

## WORKAROUND
As a workaround to grab any posters missing due to this problem simply disable Character and Creator images in Shoko Server's Settings -> AniDB -> Download Options and then refresh the AniDB Info for the effected series. You can then re-enable Character and Creator images.

**Comment by da3dsoul** [comment]
I see the issue. I'll fix it sometime today

**Comment by da3dsoul** [comment]
Fixed in 837c647493b6d375f910eb812f43b925a4fcb99b
